flang/include/flang/Optimizer/Dialect/FIRType.h

```diff
@@ -217,27 +191,12 @@ inline bool singleIndirectionLevel(mlir::Type ty) {
 }
 #endif
 
-/// Return true iff `ty` is the type of a POINTER entity or value.
-/// `isa_ref_type()` can be used to distinguish.
-bool isPointerType(mlir::Type ty);
-
 /// Return true iff `ty` is the type of an ALLOCATABLE entity or value.
 bool isAllocatableType(mlir::Type ty);
 
-/// Return true iff `ty` is the type of an unlimited polymorphic entity or
-/// value.
-bool isUnlimitedPolymorphicType(mlir::Type ty);
-
 /// Return true iff `ty` is a RecordType with members that are allocatable.
 bool isRecordWithAllocatableMember(mlir::Type ty);
 
-/// Return true iff `ty` is a RecordType with type parameters.
-inline bool isRecordWithTypeParameters(mlir::Type ty) {
```

Comment:
This one could be kept and used here: https://github.com/flang-compiler/f18-llvm-project/blob/0c82c435473327410db6944ede6cfaf42cc15eac/flang/include/flang/Optimizer/Builder/BoxValue.h#L508

Reply:
I kept this one and made the change in `BoxValue.h`. Thanks for the suggestion @jeanPerier.